hudi-spark/src/main/scala/org/apache/hudi/AvroConversionUtils.scala

```diff
@@ -96,4 +98,16 @@ object AvroConversionUtils {
     val name = HoodieAvroUtils.sanitizeName(tableName)
     (s"${name}_record", s"hoodie.${name}")
   }
+
+  private def deserializeRow(encoder: ExpressionEncoder[Row], internalRow: InternalRow): Row = {
+    // TODO remove reflection if Spark 2.x support is dropped
+    if (SPARK_VERSION.startsWith("2.")) {
```

Comment:
This was discussed here before? https://github.com/apache/hudi/pull/1760#issuecomment-685998284

using reflection in the fast path, will cause perf issues?

Reply:
Done. Added a new module ```hudi-spark3``` to avoid using reflection.